### DISCUSSION TITLE
Width of Visualization in form card

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1970] [Q&A] [answered]
Hi, I'm using ui.visualization() to plot graphs in a form card. When I don't specify a width, the visualization is barely visible as shown below

![image](https://github.com/h2oai/wave/assets/117831190/94df0e84-6575-4e50-a2c6-903d0ba1d792)

When I set the width, its visible as required. Since the default width is 100%, shouldn't it be visible?
Currently I have a ui.visualization() and a ui.expander within the form card. But the issue persists even if I create aa form_card with only the visualization.

**Comment by mturoci** [comment]
Hey, can you add a min repro code?

**Reply by n-srinidhi** [reply]
```
style ='''
div[data-test="rec_hour"].div {
  background-color: black;
}
'''

q.page['meta']=ui.meta_card(
    box='',
    title='Centra',
    stylesheet=ui.inline_stylesheet(style),
    layouts=[
        ui.layout(
            breakpoint='xs',
            zones=[
                ui.zone(name='header'),
                ui.zone(name='content',
                    zones=[
                    ui.zone(name='top'),
                    ui.zone(name='bottom')
                    ])
                ]
                ),
            ui.layout(
            breakpoint='xl',
            zones=[
                ui.zone(name='header'),
                ui.zone(name='content',
                    zones=[
                    ui.zone(name='top'),
                    ui.zone(name='bottom')
                                ])
                ])
            ]
            )

db_mstsym=data('Symbol TradesDone',size=5,
                            rows=[['BP', 298], ['ETHTHB', 147], ['AUDNZD', 128], ['GOLD_CASH', 113], ['AUDCAD', 94]],
                            pack=True)

q.page['test_section1']=ui.form_card(
            box=ui.box(zone='top'),
            items=[
                ui.inline(
            items=[
                ui.visualization(
                    data=db_mstsym,
                    plot=ui.plot([ui.mark(type='interval', x='=TradesDone', y='=Symbol', y_min=0,color=next_color())]),
                    #height='190px',
                    #width='80%',
                    ),
                  ],
                ),
                ])
```

`db_mstsym` is data_buffer

**Reply by mturoci** [reply]
Please provide a runnable minimal code example. Something I can copy paste and run directly. The above is missing data and your external stylesheet.

**Reply by n-srinidhi** [reply]
@mturoci Hi!I've edited the code. Can you please check now?

**Reply by n-srinidhi** [reply]
@mturoci Hi, Any luck with this? This seems to be a bug with the visible property, because when I recreate the item it works fine. Only when I have stated `visible=False` while creating the form card, and toggle the visibility property the width becomes too small as shown in the picture above.

**Reply by mturoci** [reply]
Thanks for pinging me, I missed your prev comment. If you have a single element in `ui.inline`, then it's redundant, and removing it should fix your problem. The default `100%` width does not apply to items in `ui.inline`, because we probably don't want to have 2 elements next to each other both taking up 100% - totaling 200% width.

The rule of thumb then is that if you use `ui.inline` you should specify widths as needed.

**Reply by n-srinidhi** [reply]
Hi! I'm getting this even for visualization without `ui.inline()`. So, If I have initially set the visible property to False, and the toggle it as True, This behavior is present. If its visible while initialization, and is toggles later it works as expected.

**Reply by mturoci** [reply, marked as answer]
Yes, correct. The `ui.visualization` plot tries to render so that it fills the entire parent container if it doesn't have dimensions specified. Since you initially start with `visible=False`, the container has no dimensions (0), so plot is not rendered (or very small). I advise rendering plots visible and when needed rather than initially unvisible.

**Reply by n-srinidhi** [reply]
Understood! Thanks a lot.
